src/tickable.ts

```diff
@@ -324,4 +351,38 @@ export abstract class Tickable extends Element {
     this.ticks = this.tickMultiplier.clone().multiply(ticks);
     this.intrinsicTicks = this.ticks.value();
   }
+
+  getAbsoluteX(): number {
+    if (!this.tickContext) {
+      throw new RuntimeError('NoTickContext', 'Tickable needs a TickContext assigned for an x-value.');
+    }
+    return this.tickContext.getX();
+  }
+
+  /** Attach this note to a modifier context. */
+  setModifierContext(mc?: ModifierContext): this {
+    this.modifierContext = mc;
+    return this;
+  }
+
+  /** Get `ModifierContext`. */
+  getModifierContext(): ModifierContext | undefined {
+    return this.modifierContext;
+  }
+
+  /**
+   * This is also used by things like beams and dots for positioning.
+   */
+  isRest(): boolean {
```

Comment:
What happens if we remove this? It feels semantically out of place.

Reply:
Originally it was in note.ts. I think I pulled it up into Tickable due to some TS warnings, but now that isCategory() is implemented, I think I can move it back into note.ts where it probably belongs!

However, I think the original comment is incorrect. Can you suggest a better comment?

note.ts:
```
  /**
   * Don't play notes by default, call them rests. This is also used by things like
   * beams and dots for positioning.
   */
  isRest(): boolean {
    return false;
  }
```

It says "call them rests" but the actual method returns false, so we aren't actually calling them rests. :-)

Or we can just use a simpler comment like in stavenote.ts:
```
  // Determine if current note is a rest.
  isRest(): boolean {
    return this.glyph.rest;
  }
```

There is another small "bug" that I have noted. this.glyph.rest is actually `true` or `undefined`... JS treats undefined as false in all the places we currently use isRest(), but at some point we should make sure isRest() actually returns a boolean.

Reply:
Okay, let's make this work, and deal with the abstractions after. Re: comment "Returns true if this note is a type of rest. Rests don't have pitches, but take up space in the score."